service/controller/v1/resource/certificate/desired.go

```diff
@@ -0,0 +1,93 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	prometheusclient "github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-config-controller/service/controller/v1/key"
+	"github.com/giantswarm/prometheus-config-controller/service/controller/v1/prometheus"
+)
+
+const (
+	caKey  = "ca"  // CaKey is the key in the Secret that holds the CA.
+	crtKey = "crt" // CrtKey is the key in the Secret that holds the certificate.
+	keyKey = "key" // KeyKey is the key in the Secret that holds the key.
+
+	// ServiceLabelSelector is the label selector to match master services.
+	serviceLabelSelector = "app=master"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
```

Comment:
All this did not really change. The diff is just funky because files moved to get the project into shape.